### PR TITLE
[feat] added Profile Context to the application

### PR DESCRIPTION
**Pull request by saitunguturu**
[//]: # "Feel free to customize this template and the emojis to your project's vibes"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- profile information can be accessed throughout the application using useProfile() from ProfileProvider.tsx

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

How the information currently looks on the profile page due to ProfileTest component:
image

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
-can be tested all at once
-have to watch out for differing UIDs in the user authentication table and adoptee_profiles tables in Supabase (currently manually inputting, need to make sure each user has their own row created in the adoptee_profiles table with matching UIDs instead of generating a new random one)
- ProfileTest component displays the profile information, and it is currently placed in the profile page

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @ethan-tam33

### CODE REVIEW

File: contexts/ProfileProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      console.error('Error fetching profile:', error);
+    } finally {
+      setProfileReady(true);
+    }
+  }, [userId]);
+
+  useEffect(() => {
+    console.log('Loading user:', userId);
+    loadProfile();
+  }, [loadProfile]);
```
Comment (ethan-tam33) [inline contexts/ProfileProvider.tsx]: Just remove this console.log statement and the eslint error will go away! We don't need to print things out anyways upon loading the profile.

File: tests/ProfileTest.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  return (
+    <div>
+      <div>User ID: {profileData.user_id}</div>
+      <div>First Name: {profileData.first_name}</div>
+      <div>Last Name: {profileData.last_name}</div>
+      <div>Email: {profileData.email}</div>
+      <div>Date of Birth: {profileData.date_of_birth}</div>
+      <div>Pronouns: {profileData.pronouns}</div>
+      <div>State: {profileData.state}</div>
+      <div>Veteran Status: {profileData.veteran_status ? 'true' : 'false'}</div>
```
Comment (ethan-tam33) [inline tests/ProfileTest.tsx]: can you instead cast the boolean `veteran_status` to a string using `.toString()`?

**Review by ethan-tam33 — CHANGES_REQUESTED** [review]

**Review by ethan-tam33 — APPROVED** [review]
